lib/ZnapZend.pm

```diff
@@ -427,6 +427,42 @@ my $createSnapshot = sub {
         }
     }
 
+    # remove snapshots from descendant subsystems that have the property "enabled" to "off", if the
+    # "recursive" flag is set to "on"
+    if ($backupSet->{recursive} eq 'on') {
+
+        $self->zLog->info("checking ZFS dependent datasets from '$backupSet->{src}' explicitely excluded");
+
+        # restrict the list to the datasets that are descendant from the current
+        my @dataSetList = grep /^$backupSet->{src}($|\/)/, @{$self->zZfs->listDataSets()};
+        if ( scalar @dataSetList > 0 ) {
+
+            # for each dataset: if the property "enabled" is set to "off", set the
+            # newly created snapshot for removal
+            my @dataSetsExplicitelyDisabled = ();
+            for my $dataSet (@dataSetList){
+                
+                # get the value for org.znapzend property
+                my @cmd = (@{$self->zZfs->priv}, qw(zfs get -H -s local -o value org.znapzend:enabled), $dataSet);
+                print STDERR '# ' . join(' ', @cmd) . "\n" if $self->debug;
+                open my $prop, '-|', @cmd;
+
+                # if the property does not exist, the command will just return. In this case,
+                # the value is implicit "on"
+                $prop = <$prop> || "on";
+                if ( substr($prop, 0, length('off')) eq 'off' ) {
```

Comment:
`if ( $prop eq 'off')` should be fine :)

Reply:
I have added a chomp also, because the command returns with a '\n'.